packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -32,25 +38,76 @@ impl PostgresDatabase {
         })
     }
 
-    /// Inserts a `version_id` and `base_id` into the `ids` table in the database
-    async fn insert_version_id(&self, id: &Identifier) -> Result<(), DatastoreError> {
+    /// Checks if the specified [`BaseId`] exist in the database.
+    ///
+    /// # Errors
+    ///
+    /// - [`DatastoreError`], if checking for the [`BaseId`] failed.
+    async fn contains_base_id(&mut self, base_id: &BaseId) -> Result<bool, DatastoreError> {
+        let ExistsRow { exists } =
+            sqlx::query_as(r#"SELECT EXISTS(SELECT 1 FROM base_ids WHERE base_id = $1);"#)
+                .bind(base_id)
+                .fetch_one(&self.pool)
+                .await
+                .report()
+                .change_context(DatastoreError)
+                .attach_printable("Could not check for base id")
+                .attach_printable_lazy(|| format!("{base_id:?}"))?;
+
+        Ok(exists)
+    }
+
+    /// Inserts the specified [`BaseId`] into the database.
+    ///
+    /// # Errors
+    ///
+    /// - [`DatastoreError`], if inserting the [`BaseId`] failed.
+    async fn insert_base_id(&mut self, base_id: &BaseId) -> Result<(), DatastoreError> {
+        sqlx::query(r#"INSERT INTO base_ids (base_id) VALUES ($1);"#)
+            .bind(base_id)
+            .execute(&self.pool)
+            .await
+            .report()
+            .change_context(DatastoreError)
+            .attach_printable("Could not insert base id")
+            .attach_printable_lazy(|| format!("{base_id:?}"))?;
+
+        Ok(())
+    }
+
+    /// Associates a [`BaseId`] with a [`VersionId`].
+    ///
+    /// # Errors
+    ///
+    /// - [`DatastoreError`], if inserting failed.
+    async fn insert_id(
+        &mut self,
+        version_id: VersionId,
```

Comment:
nit: 
If `base_id` is already taken by reference, is there a reason to own `version_id`?